umm-spec-lib/src/cmr/umm_spec/time.clj

```diff
@@ -4,6 +4,21 @@
   (:require
     [clj-time.core :as t]))
 
+(defn temporal-ranges
+  "Returns all the :RangeDateTimes and :SingeDateTimes contained in the given TemporalExtent record,
+   as a list of {:BeginningDateTime xxx :EndingDateTime xxx}"
+  [temporal]
+  (let [ranges (:RangeDateTimes temporal)
+        ;; when :EndsAtPresentFlag is set to true, set :EndingDateTime for each range to nil.
+        ;; otherwise, set :EndingDateTime to nil if it doesn't exist.
+        normalized-ranges (if (:EndsAtPresentFlag temporal) 
+                            (map #(assoc % :EndingDateTime nil) ranges)
+                            (map #(if (= nil (:EndingDateTime %)) (assoc % :EndingDateTime nil) %) ranges))
```

Comment:
This line seems a little strange. Why is it necessary to set the value for the key `:EndingDateTime` to `nil` if it is already `nil` or not in the map? Also, you should use `nil?` instead of `(= nil` here.